### PR TITLE
Add integration testing

### PR DESCRIPTION
**Pull request by ghallak**

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -4,7 +4,8 @@
 {minimum_otp_vsn, "22.3.4.9"}.
 
 {deps, [{aesophia, {git, "https://github.com/aeternity/aesophia.git",
-                    {tag, "v7.0.1"}}},
+                    %% Update before merge
```
Comment (radrow) [inline rebar.config:7]: Comment not to forget

File: rebar.lock [hunk 2]
```diff
@@ -9,7 +9,7 @@
   0},
  {<<"aesophia">>,
   {git,"https://github.com/aeternity/aesophia.git",
-       {ref,"ad4c341a4a18c5d2920d77a2b3b36fd73f8f2147"}},
+       {ref,"64281259a5a0af4fc6ed3f2d5bfce9d29e069ed8"}},
```
Comment (radrow) [inline rebar.lock:12]: ?

File: src/aere_msg.erl [hunk 3]
```diff
@@ -155,7 +165,7 @@ list_vars(Vars) ->
 
 -spec list_types([type_def()]) -> msg().
 list_types(Types) ->
-    TypesS = [ TName || {_, TName, _, _} <- Types],
+    TypesS = sets:to_list(sets:from_list([ TName || {_, TName, _, _} <- Types])),
```
Comment (radrow) [inline src/aere_msg.erl:168]: lists:usort?

File: src/aere_parse.erl [hunk 4]
```diff
@@ -37,17 +37,18 @@ commands() ->
...
         [ "Loads files into the REPL. Adds the last file to the scope."
-        , "Cleans user variables and functions. Unloads previously loaded files."]
+        , "Cleans user variables, functions, types and contracts."
+        , "Unloads previously loaded files."]
        }}
     , {"reload",
        {["r"], many_args, "[FILENAMES]",
         [ "Reloads given files preserving the included scope."
-        , "Cleans user variables and functions."]
+        , "Cleans user variables, functions, types and contracts."]
```
Comment (radrow) [inline src/aere_parse.erl:46]: Types? Really?

File: src/aere_parse.erl [hunk 5]
```diff
@@ -64,6 +65,11 @@ commands() ->
         , "cannot be adjusted using the put function."
         , "Cleans user variables and functions."]
        }}
+    , {"event",
+       {["ev"], consume, "SOPHIA_TYPE",
+        [ "Changes the in-REPL event type. Expects a Sophia variant type."
+        , "Cleans user variables, functions, typedefs, type scopes, and older events."]
```
Comment (radrow) [inline src/aere_parse.erl:71]: User is not aware of the difference between typedefs and typescopes. Just say "types".

Also, I think it's clearer to say "previously emitted events"

File: src/aere_repl.erl [hunk 6]
```diff
@@ -161,6 +164,16 @@ apply_command(continue, _, State = #repl_state{blockchain_state = BS}) ->
             {StackS, State}
     end.
 
+-spec set_event_type(aeso_syntax:typedef(), repl_state()) -> command_res().
+set_event_type(Type, S) ->
```
Comment (radrow) [inline src/aere_repl.erl:168]: Why not call clear_context?

File: src/aere_sophia.erl [hunk 7]
```diff
@@ -72,6 +72,8 @@ parse_body(I) ->
        end).
 parse_type(I) ->
     ?with_error_handle(aeso_parser:run_parser(aeso_parser:type(), I)).
+parse_typedef_variant(I) ->
+    ?with_error_handle(aeso_parser:run_parser(aeso_parser:typedef(variant), I)).
```
Comment (radrow) [inline src/aere_sophia.erl:76]: Why not parse anything and make validation later?

Comment (radrow) [reply src/aere_sophia.erl:76]: *any typedef

File: src/aere_parse.erl (hunk 4 above)
Comment (ghallak) [reply src/aere_parse.erl:46]: Yes, it uses `clear_context` which clears `typesdefs` and `typescope`.

File: src/aere_msg.erl (hunk 3 above)
Comment (ghallak) [reply src/aere_msg.erl]: Done here 3faf7b775ebd3d18fc9b741f8985c0fc255eaff5.

**Review by radrow — APPROVED** [review]

**Pull request merged**